### PR TITLE
[auth0] Fix refreshToken promise response definition and add optional refresh_token to TokenResponse type

### PR DESCRIPTION
**Pull request by andreafspeziale**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://auth0.com/docs/api/authentication#refresh-token
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

Basically:

- as soon as `/oauth/token` is called along with the `offline_access` `scope` and in the meanwhile it is enabled in the Auth0 dashboard the `TokenReponse` interface will include the `refresh_token` property which can be used later on in the `refreshToken` method or similar
```
{
    "access_token": "ACCESS_TOKEN",
    "refresh_token": "REFRESH_TOKEN",
    "id_token": "ID_TOKEN",
    "scope": "SOME_SCOPES offline_access",
    "expires_in": EXPIRATION,
    "token_type": "Bearer"
```
- the `refreshToken` was returning the `TokenResponse` in its callback definition but not in its promise definition (I believe it is happening in many other endpoints. It sounds strange to me)

### CODE REVIEW

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
https://github.com/auth0/node-auth0/blob/88cdaa93aaa2fb4f1b8a6e77d83667c18138cbaa/src/auth/index.js#L545
https://github.com/auth0/node-auth0/blob/88cdaa93aaa2fb4f1b8a6e77d83667c18138cbaa/src/auth/index.js#L562
@andreafspeziale thx!

**Comment by andreafspeziale** [comment]
Ready to merge
